lib/compress.js

```diff
@@ -677,7 +677,7 @@ merge(Compressor.prototype, {
                             || node instanceof AST_IterationStatement && !(node instanceof AST_For)
                             || node instanceof AST_SymbolRef
                                 && (node.undeclared()
-                                    || lvalues && node.name in lvalues
+                                    || lvalues && lvalues[node.name]
```

Comment:
It's not an optimisation - I do need the boolean value here. `lvalues` stores `true` if variable is modified (same as before), but `false` when the variable is read-only.

This is to facilitate the use case below when dealing with `var a=b; return b++ + a` where the modifying case is encountered during scanning as opposed to the assigned value. This case used to be taken care of by the post-order `has_side_effects()`